cypress/README.md

```diff
@@ -0,0 +1,603 @@
+
+For example, how it test looks with usage of common methods that have already exist.
+
+```typescript
+describe(`Example test`, () => {
+    beforeEach(() => {
+        basePage.openLocalhost(3001)
+    })
+
+    it(`Should open localhost`, () => {
+        basePage.checkUrlText(3001)
+    })
+})
+```
+
+### Create methods
+
+If you need to create a new method which can be reused in multiple tests you need to add it to the `common` folder `base.ts` file.
+
+If it's not:
+
+Create a `methods` folder in `sample/e2e/` folder and create `methods.ts` file in it. Then add your method to the `methods.ts` file. It will give you ability to use this method only in the sample where you created it.
+
+Note: Don't forget to import it in your test
+
+```typescript
+import {ExampleMethods} from "../methods/methods";
+```
+
+And then use it in your test
+
+```typescript
+describe(`Example test`, () => {
+    beforeEach(() => {
+        basePage.openLocalhost(3001)
+    })
+
+    it(`Should open localhost`, () => {
+        exampleMethods.checkSomething('Thing😎')
+    })
+})
+```
+
+<h2 align="center">Best Practices & Rules</h2>
+
+## Rules
+### File rules
+
+##### `base.ts`
+
+- Only for reusable methods
+- All methods need to be named correctly and understandable.
+    - Name should contain, validation type, element and state (optional). For example: `checkElementExist` or `fillFormField`
+    - Name should be camelCased
+- Every method should return something. It can be `: void` if we don't return anything (but it is set explicitly) or `: string | number | object` if we return some string, number or object etc. 
+- Methods need be added in strict order
+    - `public` (in the start)
+    - `protected` (between `public` and `private`)
+    - `private` (in the end)
+
+##### `selectors.ts`
+
+- All `selectors/locators` need be added to their blocks. By logic
+    - Buttons to `buttons` block
+    - Fields to `fields` block 
+    - And so on
+- If you don't see a suitable block, try thinking. That's for sure? If not, create a new one with a correct and understandable name
+- Try not to duplicate `selectors/locators`. Use the existing one
+- If you see that one of `selectors/locators` might be a common but it has incorrect name, please name it correctly `button` --> `commonButton`. Don't forget to check all the places where it used. 
+
+##### `constants.ts`
+- All `constants` need be added to their blocks. By logic
+    - Buttons to `buttons` block
+    - Fields to `fields` block 
+    - And so on
+- If you don't see a suitable block, try thinking. That's for sure? If not, create a new one with a correct and understandable name
+- Try not to duplicate `constants`. Use the existing one
+- If you see that one of `constants` might be a common but it has incorrect name, please name it correctly `buttonText` --> `commonButtonText`. Don't forget to check all the places where it used.
```

Comment:
please add some info that if we add new constant or selector better to check maybe it's already existed

Reply:
It's mentioned above `Try not to duplicate `constants`. Use the existing one`